test/README.md

```diff
@@ -0,0 +1,48 @@
+# test
+
+The `test` directory contains packages that correspond to
+packages found in the `lib` directory. This directory
+structure helps to keep tests scoped well and organized.
+There is also an `integration` directory specifically for
+integration tests. The test files themselves have suffixes
+that correspond to their type of test. The different
+types include `unit`, `integration` and `e2e`. This was
+chosen because the testing framework `bmocha` can `grep`
+on filename. For more documentation on `bmocha`, see
+[here](https://github.com/bcoin-org/bmocha).
+
```

Comment:
One of the scripts is off and we do not mention it in the `README`

```
"test-browser": "NODE_BACKEND=js bmocha --reporter spec test/**/*.js",
```

This script is misleading because it doesn't actually run the code in a browser, it just sets the
`NODE_BACKEND` to `js`. There is a way to set the `NODE_BACKEND` in `bmocha` using the
`-B` flag. See https://github.com/bcoin-org/bmocha/blob/fa388102f751f6d2debaea90308d53a67c71d78c/README.md

It is possible to run tests in the browser, see:
https://github.com/bcoin-org/bmocha/blob/fa388102f751f6d2debaea90308d53a67c71d78c/README.md#headless-chrome--browser-support

Reply:
The tests are significantly slower

```bash
$ npm install --save-dev browserify
$ ./node_modules/.bin/bmocha -H test/**/*.js
```

Reply:
If thought that the tree may not be able to run in browsers, but it looks the there is a Memory File System
that is used if `this.options.treePrefix` is `undefined`.
https://github.com/handshake-org/hsd/blob/ff133262ae7b83dbe86c4d96c55cca5451743bf0/lib/blockchain/chaindb.js#L51

We could set this up to actually run the tests in the browser and differentiate between `browser` and `NODE_BACKEND=js`

Reply:
I played around with trying to run tests in the browser. My proposal is to include `bpkg` as a `devDependency` so that it can be installed in the local `node_modules`, as it cannot be found if it is installed globally. Since `bpkg` does not have external dependencies, it should be safe to include as a `devDependency` specifically for testing

See: https://github.com/bcoin-org/bmocha/issues/6

Reply:
@tynes I think differentiating between `browser` tests and `NODE_BACKEND=js` tests is a good idea. How about we get the tests running in headless chrome, but do not trigger them on every commit? We can run them locally to check for browser compatibility. We can call the other script `test-js-backend`.

Reply:
> I played around with trying to run tests in the browser. My proposal is to include `bpkg` as a `devDependency` so that it can be installed in the local `node_modules`, as it cannot be found if it is installed globally. Since `bpkg` does not have external dependencies, it should be safe to include as a `devDependency` specifically for testing
> 
> See: [bcoin-org/bmocha#6](https://github.com/bcoin-org/bmocha/issues/6)

SGTM.

Reply:
@boymanjor Pushed changes to add `bpkg` as a `devDependency` and update the npm scripts. Tests pass for each of `test`, `test-browser` and `test-js`

Corresponding PR in `bcoin`:
https://github.com/bcoin-org/bcoin/pull/740